rules/typeerror.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const isTypecheckingProvider = node => node.type === 'Identifier' && node.name === 'check';
+
+const isTypecheckingOperator = operator => operator === 'typeof' || operator === 'instanceof';
+
+const isTypecheckingIdentifier = node => {
+	return node.type === 'Identifier' &&
+	(node.name.match(/[Ii]s[A-Z0-9][\w_]*/) !== null ||
```

Comment:
I think this is too loose. It will match anything that starts with `is`. Not everything `is` is typechecking. For example `isCookieMonster()`. Instead, I think we should hardcode every method we want to support. I also don't think we should target everything. Just the absolute most popular typechecks.